src/IO/Observer/Tags.hpp

```diff
@@ -131,6 +131,23 @@ struct H5FileLock : db::SimpleTag {
   static std::string name() noexcept { return "H5FileLock"; }
   using type = CmiNodeLock;
 };
+
+/// The name of the H5 file on disk to which all volume data is written.
+///
+/// \see observers::OptionTags::VolumeFileName
+struct VolumeFileName : db::SimpleTag {
+  static std::string name() { return "ObserversVolumeFileName"; }
```

Comment:
Why are the Tag and OptionTag `name`s different?  Particularly since I don't think the Tag names will be used for anything except observing and observing input file options is not very useful.

Maybe we should also add a default implementation of `name` for DataBox tags mirroring the one for option tags.

Reply:
I'm replying here also to your review message, since it's easier to keep track of comments in a thread.

1. I think the idea is that the option tag and the data box tag are conceptually different. The option tag defines a segment in the input file that yields an object. The databox tag gives an object meaning in the context of a parallel component. In particular, the databox tag should make sense on its own - it's just a descriptive label for an object. It _can_ be provided by an option tag, but doesn't have to be. The best example is probably `Tags::Domain`, which is commonly provided by `OptionTags::DomainCreator` but doesn't have to be. I think it's the cleanest approach to make this distinction for all option tags. In fact, while doing the refactoring I appreciated that I could choose names and such in the context of options and in the context of the databox separately.

  Please also consider this: In some cases the option tag carries a `std::unique_ptr`, e.g. `OptionTags::TimeStepper`. The databox tag `Tags::TimeStepper` can omit the pointer, since it will be provided by a derived compute item that returns a const ref. This again shows the distinction between the option tag providing the object and the databox tag labelling it in the context of the parallel component.

  Most importantly, on retrieval of a tag the user should not have to care how the tag got into the databox, just that it's there.

2. I don't feel strongly about the name `container_tag`, we could make it `databox_tag` if you prefer? Please make a suggestion.

3. Some `name`s differ because I tried to make them unique in the databox by including some variation of their namespace as a prefix. That's because I think the name is mostly to retrieve a tag by name at runtime. I don't feel strongly about that, either.

I'd appreciate if you could reach a conclusion on these matters with @nilsdeppe and @kidder at Cornell and let me know which changes to make, to make this more efficient.

Reply:
Another suggestion: we could derive the option tags from their respective simple tag. Then we don't need a `container_tag` alias.